bigtable/google/cloud/bigtable/row_data.py

```diff
@@ -439,18 +442,19 @@ def __iter__(self):
         while True:
             try:
                 response = self._read_next_response()
+                for chunk in response.chunks:
+                    if self.stop:
+                        raise StopIteration
```

Comment:
The cancellation could happen while a row is in progress, which means that `self.state != self.NEW_ROW`.  I think that we need to to separate out the `for` loop into a separate `try/except` block which doesn't `raise` a `ValueError`.

Reply:
Agreed. Another, shorter way could be setting the `self.state = self.NEW_ROW` just before raising the `StopIteration`. But we'll make another `try/except` block for now --- Done.